### PR TITLE
build(deps): Update K8s dependencies to v1.31

### PR DESCRIPTION
**Pull request by thunderboltsid**
Also update E2E config values to target Kubernetes v1.31 in new CI env

### CODE REVIEW

**Review by adiantum — APPROVED** [review]
lgtm

File: go.mod [hunk 1]
```diff
@@ -1,120 +1,128 @@
 module github.com/nutanix-cloud-native/cloud-provider-nutanix
 
-go 1.22.0
+go 1.23
```
Comment (yanhua121) [inline go.mod:3]: Does updating k8s.io to v1.31 require using go 1.23? I see the openshift/machine-api-operator repo still uses go 1.22 and updated k8s.io to v1.31.

Comment (thunderboltsid) [reply go.mod:3]: Are we using something from this operator as a library in Openshift?

Comment (thunderboltsid) [reply go.mod]: Synced with OCP folks, and we can keep the go.mod version change in a separate PR to make it easy for them to filter out that PR.

**Review by yanhua121 — APPROVED** [review]

**Pull request merged**